mongo/integration/sessions_test.go

```diff
@@ -475,6 +476,40 @@ func TestSessions(t *testing.T) {
 		assert.True(mt, limitedSessionUse, limitedSessMsg, len(ops))
 
 	})
+
+	// Regression test for GODRIVER-2533. Note that this test assumes the race
+	// detector is enabled (GODRIVER-2072).
+	mt.Run("NumberSessionsInProgress data race", func(mt *mtest.T) {
+		// Use a channel to sync a few simultaneous runs of NumberSessionsInProgress
+		// and a basic collection operation (CountDocuments).
+		ch := make(chan struct{})
+
+		// Use a WaitGroup to ensure that both goroutines are done executing before
+		// running test cleanup.
+		var wg sync.WaitGroup
+		wg.Add(2)
+
+		go func() {
+			defer wg.Done()
+
+			<-ch
+			for i := 0; i < 10; i++ {
+				_ = mt.Client.NumberSessionsInProgress()
+			}
```

Comment:
Optional: To increase the probability that these goroutines are running at the same time, consider adding a short sleep (e.g. 100μs) and increasing the `for` loop count to 100 or 1,000.